iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerMetrics.java

```diff
@@ -36,23 +37,44 @@ public class IoTConsensusServerMetrics implements IMetricSet {
   private Histogram writeStateMachineHistogram = DoNothingMetricManager.DO_NOTHING_HISTOGRAM;
   private Histogram offerRequestToQueueHistogram = DoNothingMetricManager.DO_NOTHING_HISTOGRAM;
   private Histogram consensusWriteHistogram = DoNothingMetricManager.DO_NOTHING_HISTOGRAM;
+  private static final String IOT_SYNC_LOG = Metric.IOT_SYNC_LOG.toString();
+  private Timer deserializeTimer = DoNothingMetricManager.DO_NOTHING_TIMER;
+  private Timer sortTimer = DoNothingMetricManager.DO_NOTHING_TIMER;
+  private Timer applyTimer = DoNothingMetricManager.DO_NOTHING_TIMER;
+  private static final String DESERIALIZE = "deserialize";
+  private static final String SORT = "sort";
+  private static final String APPLY = "apply";
 
   public IoTConsensusServerMetrics(IoTConsensusServerImpl impl) {
     this.impl = impl;
   }
 
   private static final String IMPL = "ioTConsensusServerImpl";
 
+  public void recordDeserializeCost(long costTimeInNanos) {
+    deserializeTimer.updateNanos(costTimeInNanos);
+  }
+
+  public void recordSortCost(long costTimeInNanos) {
+    sortTimer.updateNanos(costTimeInNanos);
+  }
+
+  public void recordApplyCost(long costTimeInNanos) {
+    applyTimer.updateNanos(costTimeInNanos);
+  }
+
   @Override
   public void bindTo(AbstractMetricService metricService) {
     bindAutoGauge(metricService);
     bindStageHistogram(metricService);
+    bindSyncLogTimer(metricService);
```

Comment:
I was wondering if we could unify this into Histogram or timer, otherwise it seems like there would be some difference in the way they are currently computed? May cause the time does not correspond exactly, right? @SpriCoder What's your opinion?